packages/backend-app-api/api-report.md

```diff
@@ -83,6 +85,17 @@ export const permissionsFactory: (
   options?: undefined,
 ) => ServiceFactory<PermissionsService>;
 
+// @public (undocumented)
+export const rootHttpRouterFactory: (
+  options?: RootHttpRouterFactoryOptions | undefined,
+) => ServiceFactory<RootHttpRouterService>;
+
+// @public (undocumented)
+export type RootHttpRouterFactoryOptions = {
+  indexPath?: string | false;
+  middleware?: Handler[];
```

Comment:
Is it important to be able to add middleware both at the top and the bottom of the root chain?

Reply:
Yep at some point probably, I'm thinking we'll get a better idea of that as we break out the service building and start putting that middleware straight into the service implementation.

Reply:
Mm. I'm thinking it might happen pretty soon right? With like permissions/auth going at the top and errors going at the bottom.

Reply:
Yep, thinking we'll add the ones at the end through something like ´postMiddleware´, `fallthroughMiddleware` or `errorMiddleware`